src/main/java/org/opensearch/knn/indices/ModelMetadata.java

```diff
@@ -64,6 +67,11 @@ public ModelMetadata(StreamInput in) throws IOException {
         // which is checked in constructor and setters
         this.description = in.readString();
         this.error = in.readString();
+        if (IndexUtil.isClusterOnOrAfterMinRequiredVersion(IndexUtil.MODEL_NODE_ASSIGNMENT_KEY)) {
```

Comment:
What happens in the case where the upgrade finishes, but the model was created in an old version? Will this ever lead to a break?

Reply:
I think I should add a null check after the if statement to make sure that it's set to an empty String if readOptionalString() returns nothing

Reply:
Or I can set the default value to "" in the instance variable definition

Reply:
So I think there is a bigger concern here. If we are on a new cluster version, and the metadata is created from an older version, then `in.readOptionalString` will potentially read past the stream. Because internally, `in.readOptionalString` will read a byte to see if it is there and then if so, read the whole string. However, in the older version, we dont have this byte serialized at all. I wonder if there is a hook for upgrading cluster metadata that we will need to add.

Reply:
I see this class is used for something similar: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/cluster/metadata/MetadataIndexUpgradeService.java#L60-L69C1. Can you check if there is an extension point?